nltk/tokenize/punkt.py

```diff
@@ -1321,8 +1321,8 @@ def span_tokenize(self, text, realign_boundaries=True):
         slices = self._slices_from_text(text)
         if realign_boundaries:
             slices = self._realign_boundaries(text, slices)
-        for sl in slices:
-            yield (sl.start, sl.stop)
+        for sentence in slices:
+            yield (sentence.start, sentence.stop)
```

Comment:
You use `sentence` all around, but I think these are meant to be slices? Like, `slice` objects that correspond to a sentence. `sentence` sounds like a string or a list of strings. Also, be careful, `sentence` is used as a list of tokens elsewhere in the code: 
https://github.com/nltk/nltk/blob/a31de6c687165afe64a490053c73c8e0b20ff6db/nltk/tokenize/punkt.py#L1402-L1415